Test Isolation in Truffle 5.1.5

- [x] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue

I think truffle 5.1.5 broke test file isolation.

The same contract addresses seem to be used across multiple test files when running `$(npm bin)/truffle test`. From a deeper investigation, it looks like a fresh migration is occurring, but the addresses that are injected into the environment aren't being updated from test to test.

## Steps to Reproduce

See https://github.com/mrice32/truffle_example for a simple example (varies only slightly from `truffle init`) of the test isolation issue.

```
git clone https://github.com/mrice32/truffle_example
cd truffle_example
npm install
$(npm bin)/truffle test
```

## Expected Behavior

Tests should pass and print different addresses.

## Actual Results

Output for `$(npm bin)/truffle test` in the above repository.
```

Compiling your contracts...
===========================
> Compiling ./contracts/Migrations.sol



  Contract: 1
0x8CdaF0CD259887258Bc13a92C0a6dA92698644C0
    ✓ Check and Set value (69ms)

  Contract: 2
0x8CdaF0CD259887258Bc13a92C0a6dA92698644C0
    1) Check and Set value
    > No events were emitted


  1 passing (271ms)
  1 failing

  1) Contract: 2
       Check and Set value:

      AssertionError: expected '1' to equal '0'
      + expected - actual

      -1
      +0
      
      at Context.<anonymous> (test/2.js:7:12)
      at processTicksAndRejections (internal/process/task_queues.js:86:5)
```

## Environment

* Operating System: Mac OS X 10.15
* Ethereum client: Default truffle test client
* Truffle version (`truffle version`): Truffle v5.1.5
* node version (`node --version`): v11.15.0
* npm version (`npm --version`): 6.13.4

--- a/packages/core/lib/testing/testrunner.js
+++ b/packages/core/lib/testing/testrunner.js
@@ -53,11 +53,15 @@ TestRunner.prototype.initialize = async function() {
   );
 
   if (this.first_snapshot) {
+    debug("taking first snapshot");
     try {
       let initial_snapshot = await this.snapshot();
       this.can_snapshot = true;
       this.initial_snapshot = initial_snapshot;
-    } catch (_) {}
+    } catch (error) {
+      debug("first snapshot failed");
+      debug("Error: %O", error);
+    }
     this.first_snapshot = false;
   } else {
     await this.resetState();
@@ -91,9 +95,11 @@ TestRunner.prototype.deploy = async function() {
 
 TestRunner.prototype.resetState = async function() {
   if (this.can_snapshot) {
+    debug("reverting...");
     await this.revert(this.initial_snapshot);
     this.initial_snapshot = await this.snapshot();
   } else {
+    debug("redeploying...");
     await this.deploy();
   }
 };
@@ -221,6 +227,8 @@ TestRunner.prototype.rpc = async function(method, arg) {
   if (result.error != null) {
     throw new Error("RPC Error: " + (result.error.message || result.error));
   }
+
+  return result;
 };
 
 module.exports = TestRunner;

